tests/fixtures/typescript/basics/class-multi-line-keyword-abstract.src.ts

```diff
@@ -0,0 +1,2 @@
+abstract
```

Comment:
Sorry to be a pain, but I don't think these examples of invalid TypeScript belong in basics/

Reply:
this is valid syntax :>

Expression(abstract) + class

Reply:
https://astexplorer.net/#/gist/1a4020aad417cb69813dd621c8ebec3a/b4fc915222a296b053a6b352adf0c00351c8e749

Reply:
i think i should just change name of test, but this is regression test,

we are often reading tokens before current node to check stuff, and this is just to prevent misalignment

Reply:
https://github.com/Microsoft/TypeScript/blob/master/tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractSingleLineDecl.ts

Reply:
Hmm this is less valid than many things in the errorRecovery directory IMO.

The lack of syntax/parse errors isn't the distinction we've been using.

```ts
abstract
class B {}
```

Will turn into 

```ts
"use strict";
abstract;
class B {
}
```

Which will throw

https://typescript-play.js.org/#code/IYIwzgLgTsDGEChYBthjAAgEIYN4F8g

Reply:
this is valid in non strict mode.. :>

Reply:
https://typescript-play.js.org/?alwaysStrict=false&target=1#code/IYIwzgLgTsDGEChYBthjAAgEIYN4F8g

Reply:
How? You will get a ReferenceError?

Reply:
https://typescript-play.js.org/?alwaysStrict=false#code/MYewdgzgLgBAlmKBTATgMwIbCTAvDACgEo8A+GAbwF8AoBZdLJGtEEGG6uxVTbYoA

Reply:
i can move this, or create one more with `"use strict";`

Reply:
Again, TS -> `cannot find name foo`, JS -> `throws a ReferenceError`

Reply:
I'm sorry I'm still not understanding the "use strict" role here, I must be missing something

Reply:
https://typescript-play.js.org/#code/MYewdgzgLgBAhgI2gJzsWBeGAmAUIlNKXYAGzgghgCEYBvAXyA

Reply:
in strict mode you are not allowed to use reserved keywords. in non strict mode you can use them as you want...

Reply:
i know this is super weird, but thats how its working

Reply:
Ok I see what you mean - I focusing on the wrong thing. My point was more around, if this was correct TS, abstract wouldn't be in the output at all, so that's why it seems weird to classify it as basics/

Reply:
It's an arbitrary distinction though, I admint

Reply:
ok, than do you want me to change something here?

Reply:
Nah, I'll just merge it :)